### PR TITLE
feat(textlabel component): textlabel component

### PR DESCRIPTION
**Pull request by omarVodiaK**

### CODE REVIEW

File: lib/src/textLabel/textLabel.component.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div class="text-label" [ngClass]="className">
```
[the remark this answers is not in the document]
Comment (yousifalraheem) [reply lib/src/textLabel/textLabel.component.html]: You can use bootstrap CSS variable, please don't add this class since it introduces `!important`

File: projects/ng-components-docs/src/app/examples/components/textLabels/text-labels.module.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                            description: "Additional description for example (optional)",
+                            sources: [
+                                {
+                                    name: "text-labels.component.html",
+                                    src: require("!raw-loader!./text-labels.component.html"),
+                                    lang: "markup",
+                                },
+                                {
+                                    name: "text-labels.component.ts",
+                                    // @ts-ignore
```
Comment (yousifalraheem) [inline projects/ng-components-docs/src/app/examples/components/textLabels/text-labels.module.ts]: why is this here? what's to be ignored?

File: projects/ng-components-docs/src/app/examples/components/textLabels/text-labels.component.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<div class="route-template">
+    <div class="info-holder">
+        <div class="info">
+            <div class="md-file"></div>
+        </div>
+
+        <div class="info">
+            <h2>Output</h2>
```
[the remark this answers is not in the document]
Comment (yousifalraheem) [reply projects/ng-components-docs/src/app/examples/components/textLabels/text-labels.component.html]: Agree, we don't need the old classes since this uses Storybook, just check the other examples as a reference

File: lib/src/textLabel/textLabel.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.html]: yea no way mb-1 will not be useful in this case.

File: lib/src/textLabel/textLabel.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    encapsulation: ViewEncapsulation.None,
+})
+export class TextLabelComponent {
+    @Input() id?: string;
+    @Input() label?: string | TemplateRef<HTMLElement>;
+    @Input() value?: string | TemplateRef<HTMLElement>;
+    @Input() className?: string | Array<string> | { [key: string]: boolean };
+
+    constructor() {
+        console.log("here");
```
Comment (DanSallau) [inline lib/src/textLabel/textLabel.component.ts]: The constructor here will be removed right ?

File: lib/src/textLabel/textLabel.component.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    /**
+     * Check if input parameter is a string
+     */
+    isString = (input: any): boolean => {
+        return typeof input == "string";
```
Comment (DanSallau) [inline lib/src/textLabel/textLabel.component.ts]:
```suggestion
        return typeof input === "string";
```

File: lib/src/textLabel/textLabel.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    beforeEach(() => {
+        fixture = TestBed.createComponent(TextLabelComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+});
```
Comment (DanSallau) [inline lib/src/textLabel/textLabel.component.spec.ts]: I think your test here didn't cover a scenario where either label or value is string.

File: lib/src/textLabel/textLabel.component.ts (hunk 5 above)
Comment (mario-subo) [reply lib/src/textLabel/textLabel.component.ts]: Nice catch!

File: lib/src/textLabel/textLabel.component.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<div class="text-label" [ngClass]="className" id="id">
+    <ng-container *ngTemplateOutlet="isString(label) ? defaultLabel : label"></ng-container>
+    <ng-container *ngTemplateOutlet="isString(value) ? defaultValue : value"></ng-container>
```
Comment (yousifalraheem) [inline lib/src/textLabel/textLabel.component.html:3]: Don't you think we should replace string rendering with `ng-content`?

Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.html:3]: `*ngTemplateOutlet` gives us some more power of showing the default template if none is provided. You cannot conditionally render the content within `ng-content`. You have to show the content that is received from the parent with no means to make decisions based on the content, which is what we want to achieve now.

File: lib/src/textLabel/textLabel.component.spec.ts (hunk 6 above)
Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.spec.ts:75]: added more unit tests

File: lib/src/textLabel/textLabel.component.ts (hunk 5 above)
Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.ts]: fixed

File: lib/src/textLabel/textLabel.component.ts (hunk 4 above)
Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.ts]: removed the constructor

File: lib/src/textLabel/textLabel.component.html [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<div class="text-label" [ngClass]="className">
+    <label *ngIf="label" class="custom-label" for="name">{{ label }}</label>
```
[the remark this answers is not in the document]
Comment (omarVodiaK) [reply lib/src/textLabel/textLabel.component.html]: removed

File: lib/src/textLabel/textLabel.component.html (hunk 7 above)
Comment (DanSallau) [reply lib/src/textLabel/textLabel.component.html:3]: I understand that we can achieve conditional ng-content with *ngTemplateOutlet and from your comment above i can see another different between the two. But I am a little bit skeptical about adoption/implementation/usage though as far as users are concern . Do you thing the users are familiar with *ngTemplateOutlet  and it usage ?
